src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsVocabularyAnnotation.cs

```diff
@@ -577,5 +575,31 @@ private IEnumerable<IEdmOperation> FindParameterizedOperationFromList(IEnumerabl
 
             return matchingOperations;
         }
+
+        private static int TryCount<T>(IEnumerable<T> source, out T value)
+        {
+            //// TODO check build from https://github.com/OData/odata.net/pull/2237
+            //// TODO benchmark this
+            //// TODO ensure that there are existing tests covering this
+            //// TODO add prepend stuff so that you are sure to enumerate only once?
+            using (var enumerator = source.GetEnumerator())
+            {
+                int length;
+                for (length = 0; length < 2 && enumerator.MoveNext(); ++length)
+                {
+                }
+
+                if (length == 1)
+                {
+                    value = enumerator.Current;

```

Comment:
If length ==1, enumerator.Current will be default/null, right? If length is 1 , we are trying to return the first funtion, right.